### ISSUE TITLE
Server infrastructure of GO

### ISSUE THREAD

**Issue by bjonnh** [geneontology/helpdesk#273]
I'm developing a tool "similar" to Gene Ontology but for a different domain and of much smaller scale. And I would like to get an idea about hardware stack requirements so I don't try to over-engineer it (so obviously I will scale down from the GO needs).

I would like to know if you could spare some details on the infrastructure of GO? Especially:

- What kind of machine (RAM, cores) is behind AMIGO (both for SOLR  and for the application)
- What is your SOLR architecture? Cloud or single instance?
- How much memory do you use for the reasoning tasks?

I guess Noctua requirements are pretty light, but if they are large do you mind giving some details?

**Comment by suzialeksander** [comment]
@kltm would you have anything to add here?

**Comment by kltm** [comment]
@bjonnh Hello again!

To answer what you have here:

* AmiGO itself does almost nothing and requires little. The Solr store backing it runs on a quite large machine for various reasons not strictly related to performance (128G), a secondary instance that we often use as the main instance has only 10G available and it runs without particular complaint. From experience, Solr can be flexible.
* We have various instances, both local hardware and AWS instances. We used to have a more complicated architecture, with load balancing, etc., but we currently run a single main instance for any given use.
* The answer for this is: for the data pipeline, the more the better. For building the Solr index, we supply 128G to the Solr instance and 192G to the owltools loader for Solr. The ontology build is given 48G, GO-CAM model work is given 128G. See https://github.com/geneontology/pipeline/blob/snapshot/Jenkinsfile for our builds.
* Noctua clients and middleware is quite small; Minerva--the part of Noctua that handles reasoning and the models--is typically given between 64G and 128G in our various production systems. Last time I checked in, it can realistically be run 32G and north on a laptop; reduce ontologies, etc., and it can run with less.
